radio/src/targets/horus/lcd_driver.cpp

```diff
@@ -295,66 +277,60 @@ void LCD_Init_LTDC()
+    clkConfig.PLLSAIDivR = RCC_PLLSAIDIVR_4;
   #endif
-  /* Enable PLLSAI Clock */
-  RCC_PLLSAICmd(ENABLE);
-
-  /* Wait for PLLSAI activation */
-  while(RCC_GetFlagStatus(RCC_FLAG_PLLSAIRDY) == RESET)
-  {
-  }
+  HAL_RCCEx_PeriphCLKConfig(&clkConfig);
 
   /* LTDC Configuration *********************************************************/
   /* Polarity configuration */
   /* Initialize the horizontal synchronization polarity as active low */
-  LTDC_InitStruct.LTDC_HSPolarity = LTDC_HSPolarity_AL;
+  hltdc.Init.HSPolarity = LTDC_HSPOLARITY_AL;
   /* Initialize the vertical synchronization polarity as active low */
-  LTDC_InitStruct.LTDC_VSPolarity = LTDC_VSPolarity_AL;
+  hltdc.Init.VSPolarity = LTDC_VSPOLARITY_AL;
   /* Initialize the data enable polarity as active low */
-  LTDC_InitStruct.LTDC_DEPolarity = LTDC_DEPolarity_AL;
+  hltdc.Init.DEPolarity = LTDC_DEPOLARITY_AL;
   /* Initialize the pixel clock polarity as input pixel clock */
-  LTDC_InitStruct.LTDC_PCPolarity = LTDC_PCPolarity_IPC;
+  hltdc.Init.PCPolarity = LTDC_PCPOLARITY_IPC;
 
   /* Configure R,G,B component values for LCD background color */
-  LTDC_InitStruct.LTDC_BackgroundRedValue = 0;
-  LTDC_InitStruct.LTDC_BackgroundGreenValue = 0;
-  LTDC_InitStruct.LTDC_BackgroundBlueValue = 0;
+  hltdc.Init.Backcolor.Red = 0;
+  hltdc.Init.Backcolor.Green = 0;
+  hltdc.Init.Backcolor.Blue = 0;
 
   /* Configure horizontal synchronization width */
-  LTDC_InitStruct.LTDC_HorizontalSync = HSW;
+  hltdc.Init.HorizontalSync = HSW;
   /* Configure vertical synchronization height */
-  LTDC_InitStruct.LTDC_VerticalSync = VSW;
+  hltdc.Init.VerticalSync = VSW;
   /* Configure accumulated horizontal back porch */
-  LTDC_InitStruct.LTDC_AccumulatedHBP = HBP;
+  hltdc.Init.AccumulatedHBP = HBP;
   /* Configure accumulated vertical back porch */
-  LTDC_InitStruct.LTDC_AccumulatedVBP = VBP;
+  hltdc.Init.AccumulatedVBP = VBP;
   /* Configure accumulated active width */
-  LTDC_InitStruct.LTDC_AccumulatedActiveW = LCD_PHYS_W + HBP;
+  hltdc.Init.AccumulatedActiveW = LCD_PHYS_W + HBP;
   /* Configure accumulated active height */
-  LTDC_InitStruct.LTDC_AccumulatedActiveH = LCD_PHYS_H + VBP;
+  hltdc.Init.AccumulatedActiveH = LCD_PHYS_H + VBP;
   /* Configure total width */
-  LTDC_InitStruct.LTDC_TotalWidth = LCD_PHYS_W + HBP + HFP;
+  hltdc.Init.TotalWidth = LCD_PHYS_W + HBP + HFP;
   /* Configure total height */
-  LTDC_InitStruct.LTDC_TotalHeigh = LCD_PHYS_H + VBP + VFP;
+  hltdc.Init.TotalHeigh = LCD_PHYS_H + VBP + VFP;
 
-  LTDC_Init(&LTDC_InitStruct);
+  HAL_LTDC_Init(&hltdc);
 
   // Configure IRQ (line)
-  NVIC_InitTypeDef NVIC_InitStructure;
-  NVIC_InitStructure.NVIC_IRQChannel = LTDC_IRQn;
-  NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = LTDC_IRQ_PRIO;
-  NVIC_InitStructure.NVIC_IRQChannelSubPriority = 0;
-  NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
-  NVIC_Init( &NVIC_InitStructure );
+  HAL_NVIC_SetPriority(LTDC_IRQn, LTDC_IRQ_PRIO, 0);
```

Comment:
The rest of the code uses directly the `NVIC_xxx` API, which is defined directly by CMSIS.